app/views/charts/_form.html.erb

```diff
@@ -14,22 +14,32 @@
       <div class="well">
         <%= f.input(:title) %>
         <%= f.input(:background_color, input_html: { class: :minicolors } ) %>
-        <%= f.input(:columns, collection: (1..100), prompt: true) %>
+        <%= f.input(:columns, input_html: { value: f.object.columns || 10 }) %>
+        <%= f.input(:item_height, input_html: { value: f.object.item_height || 20 }) %>
+        <%= f.input(:item_width, input_html: { value: f.object.item_width || 20 }) %>
+        <%= f.input(:width, input_html: { value: f.object.width || 1000 }) %>
+        <%= f.input(:height, input_html: { value: f.object.height || 1000 }) %>
         <%= f.input(:grouplabels) %>
-        <%= f.input(:item_height) %>
-        <%= f.input(:item_width) %>
-        <%= f.input(:file_type, collection: [['PNG', :png], ['SVG', :svg]], prompt: true) %>
-        <%= f.input(:width) %>
-        <%= f.input(:height) %>
+        <% if @chart.persisted? %>
+          <%= f.input(:file_type, collection: [['PNG', :png], ['SVG', :svg]]) %>
```

Comment:
You'll need to assign the parameters to self. It's just a regular function. Your function only returns a hash, it has no effect on the model instance.

```ruby
  after_initialize :vanilla_params, if: :new_record?
  def vanilla_params
    self.background_color = 'white'
  end
```